### PR TITLE
refactor(storage): remove uploader directly dependency of new state store

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

as title

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- remove `Uploader` directly dependency of `StorageCore`
- shared `ReadVersion` between `StorageCore` and `EventHandler`

## Checklist

- ~[ ] I have written necessary rustdoc comments~
- ~[ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- https://github.com/risingwavelabs/risingwave/issues/5621

### CODE REVIEW

File: src/storage/src/hummock/event_handler/hummock_event_handler.rs [hunk 1]
```diff
@@ -94,12 +86,23 @@ pub struct HummockEventHandler {
...
 
 impl HummockEventHandler {
     pub fn new(
         local_version_manager: Arc<LocalVersionManager>,
         shared_buffer_event_receiver: mpsc::UnboundedReceiver<HummockEvent>,
+    ) -> Self {
+        Self::new_with_read_version(local_version_manager, shared_buffer_event_receiver, None)
+    }
+
+    pub fn new_with_read_version(
```
Comment (wenym1) [inline src/storage/src/hummock/event_handler/hummock_event_handler.rs:102]: In our first intermediate stage, there will be only one read version, and the event handle should also hold the read version. So I think there is no case for passing a None read version.

**Review by wenym1 — APPROVED** [review]
Rest LGTM

File: src/storage/src/hummock/store/state_store.rs [hunk 2]
```diff
@@ -510,21 +514,29 @@ impl StateStore for HummockStorage {
         async move {
             let epoch = write_options.epoch;
             let table_id = write_options.table_id;
-            let uploader = self.core.uploader.clone();
+            // let uploader = self.core.uploader.clone();
```
Comment (hzxa21) [inline src/storage/src/hummock/store/state_store.rs:517]: nits: remove this?

File: src/storage/hummock_test/src/test_utils.rs [hunk 3]
```diff
@@ -144,9 +146,52 @@ pub async fn prepare_local_version_manager(
...
+                local_version_manager.get_pinned_version(),
+            ))),
+        )
+        .start_hummock_event_handler_worker(),
     );
 
     local_version_manager
 }
+
+pub async fn prepare_local_version_manager_new(
```
Comment (wenym1) [inline src/storage/hummock_test/src/test_utils.rs:162]: What's the difference between this function and the original `prepare_local_version_manager`?

Comment (Li0k) [reply src/storage/hummock_test/src/test_utils.rs:162]: Mainly to expose event_tx for HummockStorage::for_test.

and the function `prepare_local_version_manager_new ` not init the event_handler thread

Comment (wenym1) [reply src/storage/hummock_test/src/test_utils.rs:162]: I see. We may remove one of them after we implement the read-upload separation is finished.

**Review by hzxa21 — APPROVED** [review]
Rest LGTM.

**Pull request merged**